lib/src/metta/runner/stdlib/random.rs

```diff
@@ -66,11 +69,35 @@ impl CustomExecute for RandomFloatOp {
         if range.is_empty() {
             return Err(ExecError::from("Range is empty"));
         }
-        let mut rng = rand::thread_rng();
-        Ok(vec![Atom::gnd(Number::Float(rng.gen_range(range)))])
+        Ok(vec![Atom::gnd(Number::Float(GLOBAL_RNG.with(|generator| generator.borrow_mut().random_range(range))))])
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct SetRandomSeedOp {}
+
+grounded_op!(SetRandomSeedOp, "set-random-seed");
```

Comment:
Name of the operation should end with `!` because it has side effects.

Reply:
Done